machine/v1beta1/types_awsprovider.go

```diff
@@ -95,6 +95,14 @@ type AWSMachineProviderConfig struct {
 	// The field size should be greater than 0 and the field input must start with cr-***
 	// +optional
 	CapacityReservationID string `json:"capacityReservationId"`
+	// MarketType specifies the type of market for the EC2 instance. Valid values include:
+	// "OnDemand" (default): The instance runs as a standard OnDemand instance.
+	// "Spot": The instance runs as a Spot instance. When SpotMarketOptions is provided, the marketType defaults to "Spot".
+	// "CapacityBlock": The instance utilizes pre-purchased compute capacity (capacity blocks) with AWS Capacity Reservations.
+	//  If this value is selected, CapacityReservationID must be specified to identify the target reservation.
```

Comment:
This mentions that the `CapacityReservationID` field must be specified if this field is set to `CapacityBlock` but I don't see any validation in place for this. Use CEL validation to enforce this constraint.

For an example, see https://github.com/openshift/api/blob/c7048db20765d8169033489468c38a07679fb8ae/example/v1/types_stable.go#L131

Reply:
added